readthedocs/core/signals.py

```diff
@@ -34,52 +27,14 @@ def decide_if_cors(sender, request, **kwargs):  # pylint: disable=unused-argumen
     """
     Decide whether a request should be given CORS access.
 
-    This checks that:
-    * The URL is whitelisted against our CORS-allowed domains
-    * The Domain exists in our database, and belongs to the project being queried.
+    This checks that the URL is under ``/api/`` and it's a safe method.
 
-    Returns True when a request should be given CORS access.
+    Returns ``True`` when a request should be given CORS access.
     """
-    if 'HTTP_ORIGIN' not in request.META:
-        return False
-    host = urlparse(request.META['HTTP_ORIGIN']).netloc.split(':')[0]
-
-    # Don't do domain checking for this API for now
-    if request.path_info.startswith('/api/v2/sustainability'):
-        return True
-
-    # Don't do domain checking for APIv2 when the Domain is known
-    if request.path_info.startswith('/api/v2/') and request.method in SAFE_METHODS:
-        domain = Domain.objects.filter(domain__icontains=host)
-        if domain.exists():
-            return True
-
-    valid_url = False
-    for url in WHITELIST_URLS:
-        if request.path_info.startswith(url):
-            valid_url = True
-            break
-
-    if valid_url:
-        project_slug = request.GET.get('project', None)
-        try:
-            project = Project.objects.get(slug=project_slug)
-        except Project.DoesNotExist:
-            log.warning(
-                'Invalid project passed to domain. [%s:%s]',
-                project_slug,
-                host,
-            )
-            return False
-
-        domain = Domain.objects.filter(
-            Q(domain__icontains=host),
-            Q(project=project) | Q(project__subprojects__child=project),
-        )
-        if domain.exists():
-            return True
-
-    return False
+    return all([
+        request.method in SAFE_METHODS,
```

Comment:
We define this setting `CORS_ALLOW_METHODS`, do we really need to check this again here?

Reply:
I think these settings are confusing to me as well.

I understand that `CORS_ALLOW_METHODS` is used for the pre-flight request (`OPTIONS`) and it's to add the header `Access-Control-Allow-Methods` so after the browser can decide if will execute the final request or block it.

The signal is checked independently of these settings. If it returns `True` the origin is allowed.

In the end, `CORS_ALLOW_METHODS` is not considered when allowing it or denying it, it's just to "inform" the browser on the pre-flight request.

This chunk of code is useful to understand this behavior: https://github.com/adamchainz/django-cors-headers/blob/master/corsheaders/middleware.py#L123-L146